backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/button/BpkButtonSize.kt

```diff
@@ -0,0 +1,48 @@
+package net.skyscanner.backpack.compose.button
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.Dp
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+sealed class BpkButtonSize {
+
+  @Composable
+  internal abstract fun textStyle(): TextStyle
+
+  internal abstract val iconSize: Dp
+
+  internal abstract val minHeight: Dp
+
+  internal open val horizontalPadding: Dp =
+    BpkSpacing.Base
+
+  internal open val horizontalSpacing: Dp =
+    BpkSpacing.Md
+
+  object Default : BpkButtonSize() {
+
+    override val iconSize = BpkSpacing.Base
+
+    override val minHeight = BpkSpacing.Xl + BpkSpacing.Sm
```

Comment:
while this is true right now i'm not sure doing math with spacing tokens is a good idea long term - we might see unexpected changes when the token values change. the other part is i believe design wanted the button to scale like sp as well